code/game/barricade.dm

```diff
@@ -0,0 +1,980 @@
+					"<span class='notice'>You fumble around figuring out how to assemble [src].</span>")
+					var/fumbling_time = 10 * ( SKILL_ENGINEER_PLASTEEL - user.mind.cm_skills.engineer )
+					if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+						return
+				user.visible_message("<span class='notice'>[user] secures [src]'s anchor bolts.</span>",
+				"<span class='notice'>You secure [src]'s anchor bolts.</span>")
+				playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
+				anchored = TRUE
+				build_state = 1
+				update_icon() //unanchored changes layer
+
+			else if(iscrowbar(I))
+				if(busy || tool_cooldown > world.time)
+					return
+				tool_cooldown = world.time + 10
+				if(user.mind?.cm_skills && user.mind.cm_skills.engineer < SKILL_ENGINEER_PLASTEEL)
+					user.visible_message("<span class='notice'>[user] fumbles around figuring out how to disassemble [src].</span>",
+					"<span class='notice'>You fumble around figuring out how to disassemble [src].</span>")
+					var/fumbling_time = 50 * ( SKILL_ENGINEER_PLASTEEL - user.mind.cm_skills.engineer )
+					if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+						return
+				user.visible_message("<span class='notice'>[user] starts unseating [src]'s panels.</span>",
+				"<span class='notice'>You start unseating [src]'s panels.</span>")
+				playsound(loc, 'sound/items/crowbar.ogg', 25, 1)
+				busy = TRUE
+
+				if(!do_after(user, 50, TRUE, src, BUSY_ICON_BUILD))
+					busy = FALSE
+					return
+
+				busy = FALSE
+				user.visible_message("<span class='notice'>[user] takes [src]'s panels apart.</span>",
+				"<span class='notice'>You take [src]'s panels apart.</span>")
+				playsound(loc, 'sound/items/deconstruct.ogg', 25, 1)
+				destroy_structure(TRUE) //Note : Handles deconstruction too !
+
+
+/obj/structure/barricade/plasteel/attack_hand(mob/user as mob)
+	. = ..()
+	if(.)
+		return
+	if(isxeno(user))
+		return
+
+	playsound(src.loc, 'sound/items/ratchet.ogg', 25, 1)
+	closed = !closed
+	density = !density
+
+	if(closed)
+		user.visible_message("<span class='notice'>[user] flips [src] open.</span>",
+		"<span class='notice'>You flip [src] open.</span>")
+	else
+		user.visible_message("<span class='notice'>[user] flips [src] closed.</span>",
+		"<span class='notice'>You flip [src] closed.</span>")
+
+	update_icon()
+	update_overlay()
+
+/obj/structure/barricade/plasteel/ex_act(severity)
+	switch(severity)
+		if(1)
+			obj_integrity -= rand(450, 650)
+		if(2)
+			obj_integrity -= rand(200, 400)
+		if(3)
+			obj_integrity -= rand(50, 150)
+
+	update_health()
+
+/obj/structure/barricade/plasteel/bullet_act(obj/item/projectile/P)
+	bullet_ping(P)
+	obj_integrity -= round(P.damage/10)
+
+	if(istype(P.ammo, /datum/ammo/xeno/boiler_gas))
+		obj_integrity -= 50
+
+	update_health()
+	return TRUE
+
+/*----------------------*/
+// SANDBAGS
+/*----------------------*/
+
+/obj/structure/barricade/sandbags
```

Comment:
Please remove this one, /tg/ already has sandbags

Reply:
I think I have that now in change commit 746e6d4